tests/artifacts_tests/test_boto3.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from collections.abc import Generator
+import io
+
+import boto3
+from moto import mock_s3
+import pytest
+
+from optuna.artifacts import Boto3ArtifactStore
+from optuna.artifacts.exceptions import ArtifactNotFound
+
+
+class TestBoto3ArtifactStore:
```

Comment:
What do you think about writing these tests in a function-based manner since Optuna uses pytest runner?

Reply:
I reformat them.